Dfe.Academies.Academisation.Data/ProjectAggregate/ProjectNoteAddCommand.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Dfe.Academies.Academisation.Core;
+using Dfe.Academies.Academisation.Domain.Core.ProjectAggregate;
+using Dfe.Academies.Academisation.IData.ProjectAggregate;
+
+namespace Dfe.Academies.Academisation.Data.ProjectAggregate
+{
+	public class ProjectNoteAddCommand : IProjectNoteAddCommand
+	{
+		private readonly AcademisationContext _context;
+
+		public ProjectNoteAddCommand(AcademisationContext context)
+		{
+			_context = context;
+		}
+
+		public async Task<CommandResult> Execute(int projectId, ProjectNote note)
+		{
+			if (_context.Projects.Any(x => x.Id == projectId) is false)
```

Comment:
Why pass the Id separately and not on the ProjectNote?

Reply:
The `ProjectNote` type is the domain layer representation of the note. In the domain the relationship between a project and its notes is modelled as a collection of notes on the project, so a property for a foreign key is not required on the `ProjectNote`.

In the data layer the project note is represented with the `ProjectNoteState` type which - as it maps to rows in the db - _does_ have the ProjectId property as it maps to the foreign key column in the related table.

TL;DR - it's not in `ProjectNote` to ensure the database structure doesn't leak into the domain model.

Reply:
Perfect, makes sense.